### PR TITLE
HTTPS support

### LINKED ISSUE #101: Add support for HTTPS/secured host
Hi,

For building and testing secure sites and API, it would be great to add HTTPS support in it.

### PR DESCRIPTION
**Pull request by pavel**
* Adds HTTPS support (closes #101)

### CODE REVIEW

**Comment by tapio** [comment]
Great stuff. Would you mind creating tests for this? I assume it can be done with some self-signed snake oil certs.

I'm partial about the configuration in a js module - on the other hand it moves the complexity of reading files and handling many accepted formats in the params to the user, but then again it's not really configuration any more, but code, and you can't setup live server fully in the `~/.live-server.json`. Probably still better to leave it as is, just voicing my thoughts.

**Comment by pavel** [comment]
Sure, I'll get some tests done.

To me, personally, JSON config files are not flexible, as for some scenarios one simple value is not enough. For example take TLS, where 4 options to provide `cert`exist: a string, buffer, array of strings, or array of buffers (according to the official doc). Most people store their certs in files, which is a 5th option. And some might load a test certificate from `Let's Encrypt` service, option number 6.
**TL;DR** I'd suggest moving away from a JSON config to a JS config

**Comment by pavel** [comment]
@tapio Please, don't merge yet.

**Comment by pavel** [comment]
@tapio Seems like a problem created by this [line](https://github.com/tapio/live-server/blob/master/live-server.js#L122). After `chdir` I can no longer find HTTPS config module. Tests don't show it, because it only happens in CLI. Is it really necessary to `chdir`?

**Comment by pavel** [comment]
@tapio Yep `chdir` is definitely an issue. Mounts work only because you build their absolute pathes [here](https://github.com/tapio/live-server/blob/master/live-server.js#L85) and then use `resolve` [here](https://github.com/tapio/live-server/blob/master/index.js#L162). But this is just a coincidence, and unclear behavior. I'll fix it, if you're OK with it.

**Comment by tapio** [comment]
Are you going to change behaviour of --watch and --ignore which have been given relative to the final root (after chdir)?

**Comment by pavel** [comment]
@tapio No, that behavior will remain as was.

**Comment by pavel** [comment]
Let me add a change, so you can review it.

**Comment by tapio** [comment]
Ok. For reference, there was some discussion about mount paths here (in the line comments): https://github.com/tapio/live-server/pull/65

**Comment by pavel** [comment]
@tapio Please, review. But don't merge yet, please.

File: live-server.js [hunk 1]
```diff
@@ -116,20 +116,17 @@ for (var i = process.argv.length - 1; i >= 2; --i) {
 	}
 }
 
-if (process.argv[2]) {
-	process.chdir(process.argv[2]);
-}
-
 // Patch paths
-var cwd = process.cwd();
+var dir = opts.root = process.argv[2] || null;
```
Comment (tapio) [inline live-server.js]: This would overwrite `opts.root` given from config file. Also potentially gives null to path.join, which seems to result in TypeError.

**Comment by pavel** [comment]
About `opts.root` and `dir`.
In the previous version having call `live-server public/`, `dir`was set to `public/` watching all files there. So what's the point of `opts.root` from JSON config then?

File: live-server.js (hunk 1 above)
Comment (pavel) [inline live-server.js]: Yep, `null` is a bug.

**Comment by tapio** [comment]
The config file is for providing defaults which cli can override. So theoretically you could have "root":"public" in config and then `live-server` would always try to serve that instead of '.' unless given another path from cli. However, it seems to me that someone using it like that would be highly dubious... Also, the old behaviour is not very consistent as chdir is not performed for config file opts.root, so the path patching at the end would not produce the same result compared to when giving the root as a command line param.

**Comment by pavel** [comment]
Exactly what I was trying to say, but in more clear way.

**Comment by pavel** [comment]
I'm done from my side. You can merge, if you want to.

**Comment by tapio** [comment]
Thanks! Will do some testing and then it's time to update npm.

**Comment by pavel** [comment]
Sure. Let me know if anything is wrong after my PRs.
